### PR TITLE
Emit rerun success logs even if allow_anyone: true.

### PR DESCRIPTION
**Pull request by cjwagner**
Previously if `AllowAnyone` was true at either the global or job level we would fail to emit the "Attempted rerun" message that we use for auditing. We don't have a user name to report in this case, but it is still useful to note that the job was rerun.

/assign @BenTheElder

### CODE REVIEW

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -1409,7 +1409,7 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool, cfg
 		name := r.URL.Query().Get("prowjob")
 		l := log.WithField("prowjob", name)
 		if name == "" {
-			http.Error(w, "request did not provide the 'name' query parameter", http.StatusBadRequest)
+			http.Error(w, "request did not provide the 'prowjob' query parameter", http.StatusBadRequest)
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: heh

**Comment by cjwagner** [comment]
Should be good now. I was too impatient to wait for bazel to compile deck.

**Pull request merged**